lib/navigation/accelerometer_trajectory.cpp

```diff
@@ -0,0 +1,49 @@
+#include "accelerometer_trajectory.hpp"
+
+#include <cmath>
+
+#include <chrono>
+
+namespace hyped::navigation {
+
+AccelerometerTrajectoryEstimator::AccelerometerTrajectoryEstimator(
+  const core::ITimeSource &time, const core::TimePoint initial_time)
+    : time_(time),
+      previous_timestamp_(initial_time),
+      displacement_estimate_(0),
+      velocity_estimate_(0)
+{
+}
+
+// TODOLater: change arguments to instead take some sort of "datapoint" struct with acc_val and
+// timestamp instead
+void AccelerometerTrajectoryEstimator::update(const core::Float imu_acceleration,
+                                              const core::TimePoint imu_timestamp)
+{
+  core::Timer timer_(time_);
+  const auto time_elapsed                = timer_.elapsed(imu_timestamp, previous_timestamp_);
+  const core::Float time_elapsed_seconds = timer_.elapsedTimeInSeconds(time_elapsed);
```

Comment:
surely this class should have a private member of type `core::Timer` on which you call `elapsed` and then `reset` whenever you receive a new value

Reply:
I probably could do but wouldn't that introduce more error into the system since we would now have inconsistencies between timer values and the timestamps of the incoming data (reset uses time.now() when we would be using accelerometer data collected at a particular timepoint)? Could this be something we either suggest modifying for the honors project or play about with in the navigator benchmarker on a separate branch?